iac-extensions/arm/src/main/java/org/sonar/iac/arm/checks/CertificateBasedAuthenticationCheck.java

```diff
@@ -44,19 +44,11 @@ public class CertificateBasedAuthenticationCheck extends AbstractArmResourceChec
 
   @Override
   protected void registerResourceConsumer() {
-    register("Microsoft.ApiManagement/service/gateways/hostnameConfigurations", CertificateBasedAuthenticationCheck::checkHostnameConfigurations);
-    register("Microsoft.App/containerApps", CertificateBasedAuthenticationCheck::checkContainerApps);
-  }
+    register("Microsoft.ApiManagement/service/gateways/hostnameConfigurations", resource -> resource.property("negotiateClientCertificate")
+      .reportIf(CertificateBasedAuthenticationCheck::isBooleanFalse, DISABLED_CERTIFICATE_MESSAGE)
+      .reportIfAbsent(MISSING_CERTIFICATE_MESSAGE));
```

Comment:
Nice, I missed the feature that it was automatically embedding the name in the message!